lib/ansible/modules/cloud/amazon/ec2_vpc_vgw_facts.py

```diff
@@ -0,0 +1,169 @@
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value. See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRouteTables.html) for possible filters.
+    required: false
+    default: None
+  VpnGatewayIds:
+    description:
+      - Get details of specific Virtual Gateway ID
+      - Provide this value as a list
+    required: false
+    default: None
+author: "Nick Aslanidis (@naslanidis)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all virtual gateways for an account or profile
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+  register: vgw_facts
+
+- name: Gather facts about a filtered list of Virtual Gateways
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+    filters:
+        "tag:Name": "vgw-123"
+  register: vgw_facts
+
+- name: Gather facts about a specific virtual gateway by VpnGatewayIds
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+    VpnGatewayIds: vgw-c432f6a7 
+  register: vgw_facts
+'''
+
+RETURN = '''
+virtual_gateways:
+    description: The virtual gateways for the account
+    returned: always
+    type: list
+
+changed:
+    description: True if listing the virtual gateways succeeds
+    type: bool
+    returned: always
+'''
+
+import json
+
+try:
+    import botocore
+    import boto3   
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_virtual_gateway_info(virtual_gateway):
+    virtual_gateway_info = {'VpnGatewayId': virtual_gateway['VpnGatewayId'],
+                             'State': virtual_gateway['State'],
+                             'Type': virtual_gateway['Type'],
+                             'VpcAttachments': virtual_gateway['VpcAttachments'],
+                             'Tags': virtual_gateway['Tags']                                                              
+               }
+    return virtual_gateway_info
+
+
+def list_virtual_gateways(client, module):
+    dryrun = module.params.get("DryRun")
+    all_virtual_gateways_array = []
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filters'] = []
+        for key, value in module.params.get('filters').iteritems():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filters'].append(temp_dict)
+
+    if module.params.get("DryRun"):
+        params['DryRun'] = module.params.get("DryRun")
+
+    if module.params.get("VpnGatewayIds"):
+        params['VpnGatewayIds'] = module.params.get("VpnGatewayIds")
+
+    try:
+        all_virtual_gateways = client.describe_vpn_gateways(**params)
```

Comment:
Would it be possible for you to use the AWSRetry decorator here, so that instead of failing there will be a little bit of tolerance to the API being throttled/unavailable?